doc/source/fault-tolerance.rst

```diff
@@ -1,50 +1,91 @@
 Fault Tolerance
 ===============
 
-This document describes the handling of failures in Ray.
+This document describes how Ray handles machine and process failures.
 
-Machine and Process Failures
-----------------------------
+Tasks
+-----
 
-Each **raylet** (the scheduler process) sends heartbeats to a **monitor**
-process. If the monitor does not receive any heartbeats from a given raylet for
-some period of time (about ten seconds), then it will mark that process as dead.
+When a worker is executing a task, if the worker dies unexpectedly, either
+because the process crashed or because the machine failed, Ray will rerun
+the task (after a delay of several seconds) until either the task succeeds
+or the maximum number of retries is exceeded. The default number of retries
+is 4.
 
-Lost Objects
-------------
+You can experiment with this behavior by running the following code.
+
+.. code-block:: python
+
+    import numpy as np
+    import os
+    import ray
+    import time
+
+    ray.init(ignore_reinit_error=True)
+
+    @ray.remote(max_retries=1)
+    def potentially_fail(failure_probability):
+        time.sleep(0.2)
+        if np.random.random() < failure_probability:
+            os._exit(0)
```

Comment:
Can we use a counter actor instead of probability, so this example is deterministic?